src/overlay/FlowControl.h

```diff
@@ -54,6 +56,8 @@ class FlowControl
     size_t mDemandQueueTxHashCount{0};
     size_t mTxQueueByteCount{0};
 
+    // Mutex to synchronize flow control state
+    std::recursive_mutex mutable mFlowControlMutex;
```

Comment:
From the correctness perspective, there's no requirement to hold peer lock before acquiring the flow control lock. They're logically different locks concerned with different pieces of shared data. So I'd prefer to avoid adding an unnecessary dependency here.